source/slang/slang-ir-link.cpp

```diff
@@ -1519,6 +1520,146 @@ static void diagnoseUnresolvedSymbols(TargetRequest* req, DiagnosticSink* sink,
+        IRGLSLShaderStorageBufferType* storageBuffer;
+        {
+            IRInst* ops[] = { structType, std430 };
+            storageBuffer = builder.createGLSLShaderStorableBufferType(2, ops);
+        }
+
+        // link storage buffer to any known atomic_uint
+        instToSwitch->setFullType(storageBuffer);
+        
+        // now we need to make all call's to:
+        // 1. convert all atomic_uint function type parameters to a inlined call of atomic_uint->inout int ref of 
+        // field->element-addr based on location(offset) from the atomic_uint->storage_buffer logic above
+        for (auto& i : bindingToInstList.second)
+        {
+            int64_t currOffset = int64_t(i->findDecoration<IRGLSLOffsetDecoration>()->getOffset()->getValue());
+            // we manage the "next" ptr with a seperate list instance since we don't want to add uses
+            // of an unrelated Call to our elementAddr
+            IRUse* next = nullptr;
+            for (auto use = i->firstUse; use; use = next)
+            {
+                next = use->nextUse;
+                auto user = use->user;
+                
+                switch (user->getOp())
+                {
+                case kIROp_StructFieldLayoutAttr:
+                {
+                    // ignore the def
+                    break;
+                }
+                case kIROp_Call:
+                { 
+                    builder.setInsertBefore(user);
+                    auto fieldAddress = builder.emitFieldAddress(
+                        builder.getPtrType(_dataField->getFieldType()),
+                        instToSwitch,
+                        _dataField->getKey()
+                        );
+                    auto elementAddr = builder.emitElementAddress(
+                        builder.getPtrType(builder.getUIntType()),
+                        fieldAddress,
+                        builder.getIntValue(builder.getIntType(), currOffset/4));
+
+                    user->setOperand(1, elementAddr);
+                    auto funcTypeInst = (user->getOperand(0));
+                    auto funcType = funcTypeInst->getFullType();
+
+                    auto paramReplacment = builder.getInOutType(builder.getUIntType());
+                    funcType->getOperand(1)->replaceUsesWith(paramReplacment);
+                    builder.addForceInlineDecoration(funcTypeInst);
+
+                    break;
+                }
+                }
+            }
+            if (i->typeUse.usedValue->getOp() == kIROp_GLSLAtomicUintType)
+            {
+                i->removeAndDeallocate();
+            }
+        }
+    }
+}
+
+void tryGLSLTypeReplacmentProcessing(IRSpecContext* context, TargetProgram* targetProgram, IRModule* irModule)
+{
+    // this is an issolated boolean operation since this operation can be moved to different
+    // parts of Slang and may likley need to be moved at some point
```

Comment:
Please remove this comment.

Reply:
I will remove the comment; the idea was to explain why I have an isolated boolean since I understand it is the wrong way to go about writing code (at least in most cases, as per earlier review)